Source/Modules/javascript.cxx

```diff
@@ -459,22 +597,448 @@ int JAVASCRIPT::nativeWrapper(Node *n) {
   return SWIG_OK;
 }
 
-/* ---------------------------------------------------------------------
- * classHandler()
- *
+/**
  * Function handler for generating wrappers for class
- * --------------------------------------------------------------------- */
-
-int JAVASCRIPT::classHandler(Node *n) {
+ * When top() is called to traverse the node tree, this function is called
+ * id the node is "class" type
+ *
+ * @param n The node that the navigation is passing by
+ * @return status to swig decide what to do
+ */
+int JAVASCRIPT::classHandler(Node *n)
+{
   emitter->switchNamespace(n);
 
   emitter->enterClass(n);
-  Language::classHandler(n);
+
+  proxyClassHandlerBefore(n);
+  int returnValue = Language::classHandler(n);
   emitter->exitClass(n);
+  proxyClassHandlerAfter(n);
+
+  return returnValue;
+}
+
+/**
+ * Executed by SWIG to manage a declared C++ public variable
+ *
+ * @param n The node representing the public variable
+ * @return status to swig decide what to do
+ */
+int JAVASCRIPT::membervariableHandler(Node *n)
+{
+  proxyMemberVariableHandlerBefore(n);
+  return Language::membervariableHandler(n);
+}
```

Comment:
I think manually modifying (or adding) all `JAVASCRIPT` methods like this is not a great strategy, we should be able to do something more elegant using inheritance. It looks like inheriting `JAVASCRIPT` from `TSDECL` class, which is itself derived from `Language` would be a pretty simple solution to the problem, i.e. we could move `generateProxy` flag and all the new functions to this `TSDECL` class and avoid modifying the existing code at all.

It might not be very clear why `JS` should derive from `TS`, however, so maybe it would be better to do it in the other direction... but this would require more changes and I'm not sure if it's really worth it. But the first solution IMO very much is.